packages/gestalt/src/Typeahead.js

```diff
@@ -0,0 +1,351 @@
+
+  // Utility function for filtering data by value
+  const filterOriginalData = filterValue =>
+    dataRef.current.filter(item =>
+      item[searchField].toLowerCase().includes(filterValue.toLowerCase())
+    );
+
+  // Track input value
+  const [search, setSearch] = useState<string>(value);
+
+  // Track the selected item - could be used to see if someone is selecting the same thing again
+  const [selected, setSelected] = useState<OptionObject | null>(defaultItem);
+
+  const [options, setOptions] = useState<OptionObject[]>(
+    filterOriginalData(search)
+  );
+
+  // Ref to the input
+  const inputRef = useRef();
+
+  // Reference to selected option
+  // let selectedOptionRef;
+  // const getOptionRef = ref => {
+  //   selectedOptionRef = ref;
+  // };
+
+  // Option Container ref
+  // let containerRef;
+  // const getContainerRef = ref => {
+  //   containerRef = ref;
+  // };
+
+  // Handle when input is in and out of focus
+  const componentRef = useRef();
+  const [focused, setFocused] = useState<boolean>(false);
+
+  // When the menu item opens, scroll to selected item
+  // useEffect(() => {
+  //   setTimeout(() => {
+  //     // TODO: Fix scrolling calcultation
+  //     // if (selected !== null && containerRef && selectedOptionRef)
+  //     // eslint-disable-next-line no-use-before-define
+  //     // scrollIntoView(containerRef, selectedOptionRef);
+  //   }, 100);
+
+  //   // eslint-disable-next-line react-hooks/exhaustive-deps
+  // }, [focused, selectedOptionRef]);
+
+  const handleFocus = () => {
+    // Internally set focus status
+    if (componentRef.current)
+      setFocused(componentRef.current.contains(document.activeElement));
+
+    // Run focus callback
+    if (onFocus) onFocus();
+  };
+
+  const handleBlur = () => {
+    // Clear input and reset options
+    if (options.length === 0) {
+      setSearch('');
+      setOptions(dataRef.current);
+    }
+
+    // TODO: Is this the best way to hide the results on blur
+    if (document.activeElement && componentRef.current) {
+      setTimeout(() => {
+        setFocused(document.activeElement === componentRef.current);
+      }, 200);
+    }
+    // Run blur callback
+    if (onBlur) onBlur();
+  };
+
+  // Handler for when text is typed
+  // This rule is stupid
+  // eslint-disable-next-line no-shadow
+  const handleChange = ({ syntheticEvent: event, value }) => {
+    // Filter the available options using original data
+    const updatedOptions = filterOriginalData(value);
+
+    // Update the avalble options
+    setOptions(updatedOptions);
+
+    // Update the search value
+    setSearch(value);
+
+    // Run onChange callback
+    if (onChange) onChange({ event, value });
+  };
+
+  const handleClear = () => {
+    setSelected(null);
+    setSearch('');
+    setFocused(false);
+  };
+
+  // Handler for when an item is clicked
+  const handleSelect = item => {
+    setSelected(item);
+
+    setSearch(item[searchField]);
+    setFocused(false);
```

Comment:
@simkessy this is not what we want. Once a user selects an item, we do want to have the focus on the input element but not show the options list.

https://material-ui.com/components/autocomplete/ is a good example